packages/groq-builder/docs/MIGRATION.md

```diff
@@ -0,0 +1,149 @@
+# Migrating from GroqD v0.x to Groq-Builder v0.x 
+<!-- TODO: rename `Groq-Builder v0.x` to `groqd v1` throughout this document -->
+
+## Minimal Migration Example
+
+Migrating from `groqd` to `groq-builder` is straightforward, since there are few API changes.
+Here's an example of a simple `groqd` query, and the **minimum** changes required to migrate to `groq-builder`:
+
+#### Before, with `groqd`
+
+```ts
+import { q } from "groqd";
+
+const productsQuery = q("*")
+  .filterByType("product")
+  .order('price asc')
+  .slice(0, 10)
+  .grab({
+    name: q.string(),
+    price: q.number(),
+    slug: ["slug.current", q.string().optional()],
+    image: q("image").deref(),
+  });
+```
+
+#### After, with `groq-builder`
+
+```ts
+import { createGroqBuilderWithValidation } from "groq-builder";
+const q = createGroqBuilderWithValidation<any>(); // Using 'any' makes the query schema-unaware 
```

Comment:
I was thinking, if you "opt-out" of the strongly-typed  schema, then it ought to be explicit.  
If you've got a codebase with `no-any` rules, then you shouldn't be allowed to create a builder without a schema.

Reply:
I also was considering exporting a loosely-typed `q` too ... because it would be nice to simplify all these examples, and make it even easier to switch from groqd.  But I decided against it because:

- I wanted people to be explicit about using `any` (as I said above)
- I wanted to encourage people to create their own `q` so that if they eventually replace that `<any>` with `<SchemaConfig>` they'd already have all the imports working correctly
- I didn't want IDEs to be confused where to auto-import, because if they have a strongly-typed `q` then they might accidentally import the loosely-typed `q`.

Reply:
I didn't say this in the examples, but I guess I was thinking, you'd put this `export const q = createGroqBuilderWithValidation<any>():` in a file like `q.ts`.
Then you change all your `import { q } from 'groqd';` to `import { q } from './q';` instead.  

Should I make this clearer in the docs? Or do you think that's a pattern that folks should decide for themselves?